Firestore/Swift/Source/Result/DocumentReference+Result.swift

```diff
@@ -0,0 +1,252 @@
+  /// document exists, the update fails with an error. If a document already exists, it is
+  /// overwritten.
+  ///
+  /// - Parameters:
+  ///   - fields: The `Dictionary` containing the fields (expressed as `String` or `FieldPath`) and
+  ///   values with which to update the document.
+  ///   - completion: The closure to execute on successfully updating to the server or on receipt of
+  ///   an error. It will not be called while the client is offline, though local changes will be
+  ///   visible immediately.
+  ///   - result: The result of request. On success it is empty, otherwise it contains an `Error`.
+  func updateData(_ fields: [AnyHashable: Any],
+                  completion: @escaping (_ result: Result<Void, Error>) -> Void) {
+    updateData(fields, completion: mapResultCompletion(completion))
+  }
+
+  /// Delete the document referred to by this `DocumentReference` with the specified data.
+  ///
+  /// - Parameters:
+  ///   - completion: The closure to execute on successfully deleting to the server or on receipt of
+  ///   an error. It will not be called while the client is offline, though local changes will be
+  ///   visible immediately.
+  ///   - result: The result of request. On success it is empty, otherwise it contains an `Error`.
+  func delete(completion: @escaping (_ result: Result<Void, Error>) -> Void) {
+    delete(completion: mapResultCompletion(completion))
+  }
+
+  /// Reads the documents referenced by this `DocumentReference`.
+  ///
+  /// - Parameters:
+  ///   - source: The `FirestoreSource` where the results should be fetched:
+  ///     - `default`:  Fetch from the server and, if it fails, the cache.
+  ///     - `server`:  Fetch from the server only.
+  ///     - `cache`:  Fetch from the cache only.
+  ///   - completion: The closure to execute on receipt of a result.
+  ///   - result: The result of request. On success it contains the `DocumentSnapshot`, otherwise an
+  ///   `Error`.
+  func getDocument(source: FirestoreSource = .default,
+                   completion: @escaping (_ result: Result<DocumentSnapshot, Error>) -> Void) {
+    getDocument(source: source, completion: mapResultCompletion(completion))
+  }
+
+  /// Attaches a listener for this `DocumentReference` events.
+  ///
+  /// - Parameters:
+  ///   - includeMetadataChanges: Whether metadata-only changes (i.e. only
+  ///   `DocumentSnapshot.metadata` changed) should trigger snapshot events.
+  ///   - listener: The closure to execute on receipt of a result.
+  ///   - result: The result of request. On success it contains the `DocumentSnapshot`, otherwise an
+  ///   `Error`.
+  /// - Returns: The `ListenerRegistration` that can be used to remove this listener.
+  func addSnapshotListener(includeMetadataChanges: Bool = false,
+                           listener: @escaping (_ result: Result<DocumentSnapshot, Error>)
+                             -> Void)
+    -> ListenerRegistration {
+    addSnapshotListener(
```

Comment:
Though the compilation failure doesn't mention it, this likely also needs a return statement.

Reply:
```suggestion
   return  addSnapshotListener(
```